### DISCUSSION TITLE
Autocompletion for mid-keywords

### DISCUSSION THREAD

**Discussion by JoanGi** [eclipse-langium/langium#929] [Q&A] [answered]
Hi,

I've realized that autocompletion is not returning suggestion when you are at the mid-keyword, and this is not followed attribute like ID. 

I will try to explain myself. Given this grammar:
 ```
grammar MyDSL

entry MyDSL:
    (elements+=Declaration);

Declaration:
 'Title:' title=ID
 'Version:' version=NUMBER;

// Terminals
hidden terminal WS: /\s+/;
terminal ID: /[_a-zA-Z][\w_]*/;
terminal STRING: /"(\\.|[^"\\])*"|'(\\.|[^'\\])*'/;
terminal NUMBER returns number: /[0-9]+(\.[0-9]*)?/;  
```

The completion in the mid-keyword of **version** do not work (see image above in the playground). 
![Captura de pantalla 2023-02-13 a las 16 16 24](https://user-images.githubusercontent.com/5485515/218500241-64ff49f3-88c4-43e8-a8d3-a34bf983f193.png)

But if we change version for version=ID. It works:

![withIDs](https://user-images.githubusercontent.com/5485515/218500419-48976ed3-ea09-421d-bd69-5e777bdfd380.png)

To enable autocompletion during mid-keyword, a leaf of the AST is created with the temporal text written, but this does not happen with other types of terminals. I've thought it was a scope problem, but the autocompletion works just **before** to start typing the keyword. 

In my use-case this will improve the UX. Any thought about why this is happening? Thank you very much!!!

**Comment by msujew** [comment, marked as answer]
I've created an issue for this at https://github.com/langium/langium/issues/930, it's the same root cause as https://github.com/langium/langium/issues/778.

**Reply by msujew** [reply]
Nevermind, I've seen that the fix for #778 has not been released yet. When linking Langium locally it works as expected. We will release 1.1.0 this week:

![image](https://user-images.githubusercontent.com/4377073/218505860-168efd1f-4455-4cdb-b672-73d98de104dc.png)
